vmm/src/device_manager.rs

```diff
@@ -4201,12 +4209,21 @@ impl DeviceManager {
 
         let (pci_device, bus_device, virtio_device, remove_dma_handler) = match pci_device_handle {
             // No need to remove any virtio-mem mapping here as the container outlives all devices
-            PciDeviceHandle::Vfio(vfio_pci_device) => (
-                Arc::clone(&vfio_pci_device) as Arc<Mutex<dyn PciDevice>>,
-                Arc::clone(&vfio_pci_device) as Arc<Mutex<dyn BusDevice>>,
-                None as Option<Arc<Mutex<dyn virtio_devices::VirtioDevice>>>,
-                false,
-            ),
+            PciDeviceHandle::Vfio(vfio_pci_device) => {
+                for mmio_region in vfio_pci_device.lock().unwrap().mmio_regions() {
+                    self.mmio_regions
```

Comment:
It might be better to just put this in the `drop` method for vfio_pci_device.

Reply:
I thought about that, but it seemed like it broke OO convention to modify something the device manager tracks from a different crate. Especially given it's built/modified already by the device manager itself.